client/state/media/actions.js

```diff
@@ -178,12 +178,14 @@ export function createMediaItem( site, transientMedia ) {
  * @param {number} siteId site identifier
  * @param {object} mediaItem media item with updated properties
  * @param {object} data binary updated item data (to be sent to the server)
+ * @param {object} originalMediaItem original media item without updated properties
  */
-export const editMediaItem = ( siteId, mediaItem, data ) => ( {
+export const editMediaItem = ( siteId, mediaItem, data, originalMediaItem ) => ( {
```

Comment:
Is there a chance that the ID of the `originalMediaItem` and `mediaItem` will be different? Seems like we only use the `ID` from `originalMediaItem`.

Also, if we only use the ID, should we pass only the ID as an argument here?

Reply:
I added the whole object as we replace what's in store in case the request fails to make sure we go back to the previous state, see `receiveMedia` in [`updateMediaError`](https://github.com/Automattic/wp-calypso/blob/add/update-delete-media-error-handling/client/state/data-layer/wpcom/sites/media/index.js#L73).

Reply:
Oh, I missed that. Makes sense then 👍